docs/components.rst

```diff
@@ -1,2 +1,63 @@
 Components
-==========
\ No newline at end of file
+==========
+
+Everything which is required to provision a virtual machine or docker container are wrapped under a term called components.
+Read further for complete details.
+
+Infrastructure
+--------------
+This includes information regarding hardware availability of remote machine.
```

Comment:
Instead of this write something about infrastructure that it is another system in the same network, which has libvirt running or has qemu hypervisor installed..